hookman/hookman_generator.py

```diff
@@ -279,18 +287,34 @@ def generate_plugin_package(
 
         self._validate_package_folder(artifacts_dir, assets_dir)
         self._validate_plugin_config_file(assets_dir / "plugin.yaml")
-        version = PluginInfo(assets_dir / "plugin.yaml", hooks_available=None).version
+        plugin_info = PluginInfo(assets_dir / "plugin.yaml", hooks_available=None)
 
         if sys.platform == "win32":
             shared_lib_extension = "*.dll"
-            hmplugin_path = dst_path / f"{package_name}-{version}-win64.hmplugin"
+            hmplugin_path = dst_path / f"{package_name}-{plugin_info.version}-win64.hmplugin"
         else:
             shared_lib_extension = "*.so"
-            hmplugin_path = dst_path / f"{package_name}-{version}-linux64.hmplugin"
+            hmplugin_path = dst_path / f"{package_name}-{plugin_info.version}-linux64.hmplugin"
+
+        contents = (assets_dir / 'plugin.yaml').read_text()
+        if extras_defaults is not None:
+            import strictyaml
+
+            contents_dict = strictyaml.load(contents, PLUGIN_CONFIG_SCHEMA)
+            if 'extras' not in contents_dict:
+                contents_dict['extras'] = extras_defaults
+            else:
+                for key, value in extras_defaults.items():
+                    contents_dict['extras'][key] = contents_dict['extras'].get(key, value)
+            contents = contents_dict.as_yaml()
 
         with ZipFile(hmplugin_path, "w") as zip_file:
+
             for file in assets_dir.rglob("*"):
-                zip_file.write(filename=file, arcname=file.relative_to(plugin_dir))
```

Comment:
Why this difference? 🤔

Reply:
I don't want to change the original user's `plugin.yaml` file, so I'm getting its contents, manipulating  it and generating the target `plugin.yaml` file from this result. Using `writestr` because it was simpler then generating a temporary file just to use the `write` (would still need the content update and some handling in this for/if)